### ISSUE TITLE
Accordion: Dynamic headers causes tab unmount

### ISSUE THREAD

**Issue by dattebayorob** [primefaces/primereact#2319]
```
[ x ] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**

https://codesandbox.io/s/pensive-hooks-zkfmd?file=/src/demo/AccordionDemo.css:239-637

**Current behavior**

When transitionOptions option unmountOnExit is set to false, and header is dymamic updated, the tab content will unmount and remount after changes.
The issue is because in Accordion.js line 170, the header is used as key of the element. 

I think, the component can receive a prop called tabKey as used in DataTable to prevent this.

```
    renderTab(tab, index) {
        const selected = this.isSelected(index);
        const tabHeader = this.renderTabHeader(tab, selected, index);
        const tabContent = this.renderTabContent(tab, selected, index);
        const tabClassName = classNames('p-accordion-tab', {
            'p-accordion-tab-active': selected
        });
        // React.children.map adds a key implicit using the index, maybe just accepting tabKey or undefined should be ok, and no need to use header as key
        const key = tab.props.tabKey ? tab.props.tabKey : tab.props.header;
        
        return (
            <div key={key} className={tabClassName}>
                {tabHeader}
                {tabContent}
            </div>
        );
    }
```
Obs: Is possible to bypass  this, using headerTemplate prop instead of header and let React.children.map handle the key itself: https://stackoverflow.com/a/50303840

Also, when using unmountOnExit false the Tab is still visible after toggle, I was able to bypass this via css:
```
.p-accordion-tab:not(.p-accordion-tab-active)
  .p-toggleable-content:not(.p-toggleable-content-enter):not(.p-toggleable-content-enter-done):not(.p-toggleable-content-exit):not(.p-toggleable-content-exit-done)
  .p-accordion-content {
  display: none;
}
.p-accordion-tab:not(.p-accordion-tab-active)
  .p-toggleable-content.p-toggleable-content-exit-done
  .p-accordion-content {
  display: none;
}
```
@mertsincan if helps, I can open a PR to fix this.

**Comment by mertsincan** [comment]
Hi @dattebayorob,

A PR can be nice ;) Also, can you create a PR for TabView? They are of similar structure.

**Comment by melloware** [comment]
This is fixed in 8.0.0 see: https://codesandbox.io/s/romantic-sutherland-hiyutk
